addons/l10n_sg/models/res_bank.py

```diff
@@ -0,0 +1,58 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartnerBank(models.Model):
+    _inherit = 'res.partner.bank'
+
+    proxy_type = fields.Selection(selection_add=[('mobile', 'Mobile Number'), ('uen', 'UEN')],
+                                  ondelete={'mobile': 'set default', 'uen': 'set default'})
+
+    @api.constrains('proxy_type', 'proxy_value', 'partner_id')
+    def _check_sg_proxy(self):
+        for bank in self.filtered(lambda b: b.country_code == 'SG'):
+            if bank.proxy_type not in ['mobile', 'uen', 'none', False]:
+                raise ValidationError(_("The PayNow Type must be either Mobile or UEN to generate a PayNow QR code."))
```

Comment:
In the res.partner view, this error would raise when saving the res.partner itself. I think we could add the bank account on which the error occurs to the error to make it easier for the user to troubleshoot, what do you think?